### PR TITLE
refactor: static analysis fixes

### PR DESCRIPTION
**Pull request by timtebeek**
Figured apply a few quick static analysis fixes; can do more, but only if desired;
Thought to keep this small and easy to review. If not desired that's fine as well!

### CODE REVIEW

File: slf4j-api/src/main/java/org/slf4j/helpers/Util.java [hunk 1]
```diff
@@ -53,7 +53,7 @@ public static boolean safeGetBooleanSystemProperty(String key) {
         if (value == null)
             return false;
         else
-            return value.equalsIgnoreCase("true");
```
Comment (HannesWell) [inline slf4j-api/src/main/java/org/slf4j/helpers/Util.java:56]: What about just return `Boolean.parseBoolean(value)` instead?

Comment (timtebeek) [reply slf4j-api/src/main/java/org/slf4j/helpers/Util.java]: That's indeed even better in this specific case; pushed just now.

The previous iteration was made by applying [the "Equals avoids null" recipe](https://docs.openrewrite.org/reference/recipes/java/cleanup/equalsavoidsnull); Perhaps a subsequent recipe can convert that into the `Boolean.parseBoolean` calls in other places as well.

File: slf4j-api/src/main/java/org/slf4j/helpers/SubstituteLogger.java [hunk 2]
```diff
@@ -418,10 +418,7 @@ public boolean equals(Object o) {
 
         SubstituteLogger that = (SubstituteLogger) o;
 
-        if (!name.equals(that.name))
-            return false;
-
-        return true;
+        return !(!name.equals(that.name));
```
Comment (HannesWell) [inline slf4j-api/src/main/java/org/slf4j/helpers/SubstituteLogger.java:421]:
```suggestion
        return name.equals(that.name);
```

**Review by HannesWell — COMMENTED** [review]
Looks good to me in general.
But I'm not a maintainer of this project and can therefore not decide if this is considered for inclusion.

**Comment by timtebeek** [comment]
Hi @ceki, sure! I'm using the public instance of Moderne, at https://public.moderne.io
That allows you to run any of the 1000+ recipes against Open Source repositories.
Here's a quick overview: https://public.moderne.io/marketplace/org.openrewrite.java

You'll likely want to [create an organization group for qos.ch](https://public.moderne.io/settings/repo-groups), as I've done, to preview changes.
image

Then once you're satisfied you can choose to create a pull request like I've done here as well.
That way you're able to run some of the same recipes for logback as well. Here's a temporary preview link; which will only last as long as the worker containing the results: https://public.moderne.io/results/cC08T
image

This ran the fairly large [composed recipe for common static analysis issues](https://public.moderne.io/recipes/org.openrewrite.java.cleanup.CommonStaticAnalysis), but you can also run a subset to get a smaller changeset, as I had done before.
image

Hope this helps; we made the platform self-serve such that you're able to quickly explore and run recipes yourself, optionally skipping the coordination and pull requests here.

And in addition to these clean up recipes, we also have logging specific recipes for your users to migrate to logging best practices such: https://public.moderne.io/marketplace/org.openrewrite.java.logging , all based on open source recipes at https://github.com/openrewrite/rewrite-logging-frameworks/tree/main/src/main

Curious to hear your thoughts, and if there's anything we can improve or help with. :)

**Comment by xavier-figueroa** [comment]
Hi @timtebeek I was about to to similar cleanup but using IDE tools. I hope your PR gets merged soon. Moderne looks great by the way, I'll give it a shot

**Comment by timtebeek** [comment]
> Hi @timtebeek I was about to to similar cleanup but using IDE tools. I hope your PR gets merged soon. Moderne looks great by the way, I'll give it a shot

Thanks! Yes the platform makes it really easy to pitch such improvements to OSS projects, although there is a balance not to cause any review burden. That is why I typically start small and add a bit of background before doing more. Let me know if you push any nice improvements somewhere; I'd love to see that!

**Comment by timtebeek** [comment]
I'm going to close this one due to inactivity. I'll keep the fork & branch up in case it's desired still.

**Pull request closed without merging**